transformer_engine/pytorch/csrc/userbuffers/userbuffers-host.cpp

```diff
@@ -44,16 +46,33 @@ int stringCmp(const void *a, const void *b) { return strcmp((const char *)a, (co
     }                                                                                       \
   } while (0)
 
-#define CUCHECK(cmd)                                                               \
-  do {                                                                             \
-    CUresult retval = cmd;                                                         \
-    if (retval != CUDA_SUCCESS) {                                                  \
-      const char *error_string;                                                    \
-      cuGetErrorString(retval, &error_string);                                     \
-      printf("Failed: Cuda error %s:%d '%s'\n", __FILE__, __LINE__, error_string); \
-      exit(EXIT_FAILURE);                                                          \
-    }                                                                              \
-  } while (0);
+// Call-by-symbol tool for CUDA driver API -- this helps avoid linking to libcuda.so at compile
+static std::map<const char *, void *> _cudaDriverEntryPoints;
+
+template <typename... Args>
+cudaError_t cudaCallDriverEntryPoint(const char *symbol, Args... args) {
```

Comment:
probably ought to avoid defining functions with names that look like CUDA API calls that aren't.